core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowResource.scala

```diff
@@ -25,12 +25,17 @@ import java.util
 import javax.servlet.http.HttpSession
 import javax.ws.rs._
 import javax.ws.rs.core.{MediaType, Response}
+import scala.collection.JavaConverters._
+import scala.collection.mutable
 
 /**
   * This file handles various request related to saved-workflows.
   * It sends mysql queries to the MysqlDB regarding the UserWorkflow Table
   * The details of UserWorkflowTable can be found in /core/scripts/sql/texera_ddl.sql
   */
+
+case class workflowInfo(isOwner: Boolean, workflow: Workflow)
```

Comment:
@Jiyang-Wu this design has a big problem.

I understand you want to use a larger object to contain both workflow and ownership information. The name `WorkflowInfo` is problematic: the name has been used somewhere else. I see you used `UserFileAccess` object for file sharing, I think we need to come up with a unified way of such return response object, for both workflow + access and file + access.

On the frontend, you simply changed original `Workflow` to `WorkflowInfo` and other names and comments are not changed, which results in `this.workflow.workflow` this kind of unreadable code. This is another issue.

Reply:
Changed to `DashboardWorkflowEntry` and `DashboardUserFileEntry`.